src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -77,19 +78,33 @@ public WritableShreddedDocument shred(JsonNode doc, UUID txId) {
         txId,
         IndexingProjector.identityProjector(),
         "testCommand",
-        CollectionSettings.empty());
+        CollectionSettings.empty(),
+        null);
   }
 
   public WritableShreddedDocument shred(CommandContext ctx, JsonNode doc, UUID txId) {
-    return shred(doc, txId, ctx.indexingProjector(), ctx.commandName(), ctx.collectionSettings());
+    return shred(
+        doc, txId, ctx.indexingProjector(), ctx.commandName(), ctx.collectionSettings(), null);
+  }
+
+  public WritableShreddedDocument shred(
+      CommandContext ctx, JsonNode doc, UUID txId, AtomicReference<DocumentId> docIdToReturn) {
+    return shred(
+        doc,
+        txId,
+        ctx.indexingProjector(),
+        ctx.commandName(),
+        ctx.collectionSettings(),
+        docIdToReturn);
   }
 
   public WritableShreddedDocument shred(
       JsonNode doc,
       UUID txId,
       IndexingProjector indexProjector,
       String commandName,
-      CollectionSettings collectionSettings) {
+      CollectionSettings collectionSettings,
+      AtomicReference<DocumentId> docIdToReturn) {
```

Comment:
Can you add a comment for this new field?

Reply:
Will do.

Reply:
done.